src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -270,8 +270,11 @@ class SnykLanguageClient :
     fun hasAuthenticated(param: HasAuthenticatedParam) {
         if (disposed) return
         val oldToken = pluginSettings().token
-        if (oldToken == param.token) return
+        val oldApiUrl = pluginSettings().customEndpointUrl
+        if (oldToken == param.token && oldApiUrl == param.apiUrl) return
         pluginSettings().token = param.token
```

Comment:
this should only be overwritten, if it's unequal (empty is allowed)

Reply:
Fixed